File: docs/guides/java/src/test/java/com/algolia/searchWithGAToken.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;

public class searchWithGAToken {

  private static String getGoogleAnalyticsUserIdFromBrowserCookie(String cookieName) {
    return ""; // Implement your logic here
  }

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")) {
      String userToken = getGoogleAnalyticsUserIdFromBrowserCookie("_ga");
      SearchParamsObject searchParams = new SearchParamsObject().setQuery("<YOUR_SEARCH_QUERY>").setUserToken(userToken);

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);

      String loggedInUser = null;
      searchParams.setUserToken(loggedInUser != null ? loggedInUser : userToken);

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchWithGAToken.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*

import com.algolia.client.model.search.*

val getGoogleAnalyticsUserIdFromBrowserCookie: (String) -> String = {
  "" // Implement your logic here
}

suspend fun searchWithGAToken() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  val userToken = getGoogleAnalyticsUserIdFromBrowserCookie("_ga")
  var searchParams = SearchParamsObject(query = "<YOUR_SEARCH_QUERY>", userToken = userToken)

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)

  val loggedInUser: String? = null
  searchParams = SearchParamsObject(query = "<YOUR_SEARCH_QUERY>", userToken = loggedInUser ?: userToken)

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
